### PR TITLE
5100: new mobile styles for sonar  feed

### PR DESCRIPTION
**Pull request by GarageInc**
Done: 
* Divide projects by supported metrics(ERC-20 for Historycal balance with "wallet" address and not-ERC20(withous address))

* Desktop list
image

* Mobile view
image

### CODE REVIEW

File: src/components/SignalCard/SignalCard.module.scss [hunk 1]
```diff
@@ -91,15 +109,21 @@
...
 .expandButton {
-  color: var(--waterloo);
-  fill: var(--waterloo);
+  margin-right: 20px;
+  padding-left: 0;
+
+  @include responsive('phone', 'phone-xs') {
+    margin: 0;
+    padding: 6px 6px;
+  }
```
Comment (haritonasty) [inline src/components/SignalCard/SignalCard.module.scss]: `padding: 6px;`

File: src/components/SignalCard/SignalCard.module.scss [hunk 2]
```diff
@@ -4,11 +4,15 @@
   display: flex;
 
   &__right {
-    padding: 4px 0;
+    padding: 4px 0 0 0;
```
Comment (haritonasty) [inline src/components/SignalCard/SignalCard.module.scss]: can be `padding: 4px 0 0`

File: src/components/SignalCard/SignalCard.js [hunk 3]
```diff
@@ -149,4 +118,48 @@ const SignalCardBottom = ({
...
+  const [isOpen, setOpen] = useState(false)
+
+  return (
+    <div className={styles.more} onMouseLeave={() => setOpen(false)}>
+      <Button
+        onClick={() => setOpen(!isOpen)}
+        className={cx(styles.expandButton)}
+      >
+        <Icon type='dots' className={styles.moreIcon} />
+      </Button>
```
Comment (haritonasty) [inline src/components/SignalCard/SignalCard.js]: looks like [`ContextMenu` component](https://storybook.stage.internal.santiment.net/?selectedKind=ContextMenu&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel)
example in app: 
image

File: src/components/SignalCard/SignalCard.module.scss (hunk 2 above)
Comment (GarageInc) [reply src/components/SignalCard/SignalCard.module.scss]: Padding-master :)

File: src/components/SignalCard/SignalCard.js (hunk 3 above)
Comment (GarageInc) [reply src/components/SignalCard/SignalCard.js]: Correct, will check

File: src/components/SignalCard/SignalCard.js [hunk 4]
```diff
@@ -75,49 +82,12 @@ const SignalCardBottom = ({
   toggleSignal
 }) => {
   const isUserTheAuthor = true
```
Comment (Partysun) [inline src/components/SignalCard/SignalCard.js]: Const outside the component. to top of the file. and uppercase.
@GarageInc

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js [hunk 5]
```diff
@@ -53,25 +50,20 @@ export const TriggerFormAssetWallet = ({
...
-            if (ethAddress) {
-              if (metaFormSettings.target.value.value === newAsset.value) {
-                setFieldValue('ethAddress', ethAddress)
-              } else {
-                setFieldValue('ethAddress', '')
-              }
-            }
-          }}
-        />
+        {isEthWallet && (
```
Comment (Partysun) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: You can choose component and then use it.
const ProjectsView = isEthWallet ? ProjectByErc20 : ...

<ProjectsView

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelectorWrapper.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { graphql } from 'react-apollo'
+import { compose } from 'recompose'
```
Comment (Partysun) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelectorWrapper.js]: Compose from redux. Better.

File: src/pages/Projects/allProjectsGQL.js [hunk 7]
```diff
@@ -133,4 +133,13 @@ export const currenciesMarketSegmentsGQL = gql`
   }
 `
 
+export const ALL_ERC20_PROJECTS_QUERY = gql`
```
Comment (Partysun) [inline src/pages/Projects/allProjectsGQL.js]: @GarageInc Do we have this in our queries/ directory?
Does this query unique only for this place?

File: src/components/SignalCard/SignalCard.js [hunk 8]
```diff
@@ -4,9 +4,12 @@ import cx from 'classnames'
 import Panel from '@santiment-network/ui/Panel/Panel'
 import Icon from '@santiment-network/ui/Icon'
 import Toggle from '@santiment-network/ui/Toggle'
+import Button from '@santiment-network/ui/Button'
+import { DesktopOnly, MobileOnly } from './../../components/Responsive'
 import MultilineText from '../../components/MultilineText/MultilineText'
 import StatusLabel from './../../components/StatusLabel'
 import { RemoveSignalButton, SignalTypeIcon } from './controls/SignalControls'
+import { ContextMenu } from '@santiment-network/ui'
```
Comment (DmitriVanGuard) [inline src/components/SignalCard/SignalCard.js]: This import statement [does not match our style guide](https://github.com/santiment/san-style-guide#webpack-tree-shaking-problem)

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js (hunk 5 above)
Comment (GarageInc) [reply src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: Correct, will fix

File: src/pages/Projects/allProjectsGQL.js (hunk 7 above)
Comment (GarageInc) [reply src/pages/Projects/allProjectsGQL.js]: This query used in BalancesView component(checking wallet balance) and in TriggerForm

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelectorWrapper.js (hunk 6 above)
Comment (GarageInc) [reply src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelectorWrapper.js]: A lot of components import from recompose
image
Will change

File: src/components/SignalCard/SignalCard.js (hunk 8 above)
Comment (GarageInc) [reply src/components/SignalCard/SignalCard.js]: Fixed

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelector.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  metaFormSettings,
+  target,
+  setFieldValue
+}) => {
+  const { ethAddress, target: defaultAsset } = metaFormSettings
+
+  const options = projects.map(asset => ({
+    label: asset.slug,
+    value: asset.slug
+  }))
```
Comment (haritonasty) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelector.js]:
```
projects.map(({slug})) => ({
    label: slug,
    value: slug
  }))
```
will be better

Comment (haritonasty) [reply src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelector.js]: and why you devided to two fields? (just interest)

File: src/components/formik-santiment-ui/FormikSelect.js [hunk 10]
```diff
@@ -14,20 +14,41 @@ const FormikSelect = ({
...
+      render={({ field, form }) => {
+        const isValid = !form.errors[name]
+        const customStyles = {
+          control: (base, state) => {
+            return {
+              ...base,
+              borderColor: isValid ? '#ddd' : 'var(--persimmon) !important',
+              '&:hover': {
+                borderColor: isValid ? '#ddd' : 'var(--persimmon) !important'
+              }
```
Comment (haritonasty) [inline src/components/formik-santiment-ui/FormikSelect.js]: create variable for repeating code, you are constant person

File: src/components/SignalCard/SignalCard.js [hunk 11]
```diff
@@ -1,12 +1,15 @@
...
 import cx from 'classnames'
 import Panel from '@santiment-network/ui/Panel/Panel'
 import Icon from '@santiment-network/ui/Icon'
 import Toggle from '@santiment-network/ui/Toggle'
+import Button from '@santiment-network/ui/Button'
+import { DesktopOnly, MobileOnly } from './../../components/Responsive'
 import MultilineText from '../../components/MultilineText/MultilineText'
 import StatusLabel from './../../components/StatusLabel'
 import { RemoveSignalButton, SignalTypeIcon } from './controls/SignalControls'
+import ContextMenu from '@santiment-network/ui/ContextMenu'
```
Comment (haritonasty) [inline src/components/SignalCard/SignalCard.js]: wrong order( 
`ContextMenu` should be above

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelector.js (hunk 9 above)
Comment (GarageInc) [reply src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerProjectsSelector.js]: It used in selects, which requires 'label' and 'value' for every option.

File: src/components/SignalCard/SignalCard.js (hunk 11 above)
Comment (GarageInc) [reply src/components/SignalCard/SignalCard.js]: Fill wix

File: src/components/formik-santiment-ui/FormikSelect.js (hunk 10 above)
Comment (GarageInc) [reply src/components/formik-santiment-ui/FormikSelect.js]: Fixed

File: src/components/SignalCard/SignalCard.js [hunk 12]
```diff
@@ -149,4 +118,49 @@ const SignalCardBottom = ({
   )
 }
 
+const MoreSignalActions = ({ signalId, removeSignal }) => {
+  return (
+    <ContextMenu
+      trigger={
+        <Button className={cx(styles.expandButton)}>
```
Comment (DmitriVanGuard) [inline src/components/SignalCard/SignalCard.js]: Button has a single class. No need for a concatenation

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js [hunk 13]
```diff
@@ -77,4 +92,53 @@ export const TriggerFormAssetWallet = ({
...
+      allErc20Projects: allErc20Projects || data.allErc20Projects || [],
+      allProjects: allProjects || data.allProjects || []
+    }
+  }
+}
+
+const pickGQL = type => {
+  const defaultQuery = allProjectsForSearchGQL
+  switch (type) {
+    case 'all':
```
Comment (DmitriVanGuard) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: It's same as the `default` cause, therefore this line could omitted

**Pull request merged**